utils/build/docker/nodejs/express4/app.js

```diff
@@ -182,6 +182,82 @@ app.get('/dsm', (req, res) => {
     })
 })
 
+app.get('/kafka/produce', (req, res) => {
+  const topic = req.query.topic
+  const kafka = new Kafka({
+    clientId: 'my-app-producer',
+    brokers: ['kafka:9092'],
+    retry: {
+      initialRetryTime: 100, // Time to wait in milliseconds before the first retry
+      retries: 20 // Number of retries before giving up
+    }
+  })
+  const admin = kafka.admin()
+  const producer = kafka.producer()
+  const doKafkaOperations = async () => {
+    await admin.connect()
+    await producer.connect()
+    await admin.createTopics({
+      waitForLeaders: true,
+      topics: [
+        { topic }
+      ]
+    })
```

Comment:
do we actually need to do that ? I think just sending a message to a non-existent topic creates it. Not sure if it's a feature of kafka, or of the frameworks.
(Also, won't it fail if that endpoint is called twice ?)

Reply:
The topic _is_ created twice, with the second attempt generating a WARNING ("topic already exists").  However, without waiting for leaders the test was flaky because the client attempted to publish to the topic before the Leader was elected, generating an ERROR ("topic does not have a leader").  

This line causes gets us the WARNING to replace the ERROR, which means no more flaky test